### PR TITLE
Remove `make -jX`  step in qtcreator instructions

### PR DESCRIPTION
**Pull request by merydian**
This step is no longer necessary.

Goal: Improve instructions for building QGIS with QtCreator

Ticket(s): #

- [ ] Backport to LTR documentation is requested

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

File: docs/developers_guide/qtcreator.rst [hunk 1]
```diff
@@ -104,30 +104,6 @@ you may want to enable more features such as:
...
-
-#. Press the :menuselection:`Add build step -->` menu and select
-   :guilabel:`Custom Process Step`
-#. Fill the new form as follows:
-
-   * :guilabel:`Command`: ``make``
-   * :guilabel:`Arguments`: ``-j4`` to use 4 cores (setting depends on your device)
-   * :guilabel:`Working directory`: ``%{buildDir}``
-
-.. image:: img/customProcess.png
```
Comment (DelazJ) [inline docs/developers_guide/qtcreator.rst:119]: I think yu would also need to delete this file as it might become unused.

**Review by DelazJ — COMMENTED** [review]
Thanks for the cleanup @merydian. May I ask what is the new behavior? Qt5 or Qt6? Also is the ninja alternative tip also unnecessary?

**Comment by nyalldawson** [comment]
@DelazJ 
> Thanks for the cleanup @merydian. May I ask what is the new behavior?

Qtcreator will just always use as much resources as is reasonable now. I think this step was only needed in really old QtCreator versions. 

>  Is it in Qt5 or Qt6? 

It's for either.

> Also is the ninja alternative tip also unnecessary?

It's necessary to install ninja manually, but then QtCreator will default to it automatically if it's available.

**Comment by DelazJ** [comment]
Thanks for the explanation @nyalldawson

**Pull request merged**